### PR TITLE
docs: update "Development" guide

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

Updates the "Development" guide and makes it consistent with APISIX guide.

### CODE REVIEW

File: docs/en/latest/contribute.md [hunk 1]
```diff
@@ -21,92 +27,103 @@ title: Developing for Apache APISIX Ingress Controller
...
+This document walks through how you can set up your development environment to contribute to APISIX Ingress controller.
 
 ## Prerequisites
 
-* Install [Go 1.19](https://golang.org/dl/) or later, and we use go module to manage the go package dependencies.
-* Prepare an available Kubernetes cluster in your workstation, we recommend you to use [KIND](https://kind.sigs.k8s.io/).
-* Install Apache APISIX in Kubernetes by [Helm Chart](https://github.com/apache/apisix-helm-chart).
+Before you get started make sure you have:
+
+1. Installed [Go 1.13](https://golang.org/dl/) or later
```
Comment (tao12345666333) [inline docs/en/latest/contribute.md:36]: we should use Go 1.19

**Review by tao12345666333 — CHANGES_REQUESTED** [review]
others LGTM

File: docs/en/latest/contribute.md (hunk 1 above)
Comment (navendu-pottekkat) [reply docs/en/latest/contribute.md:36]: Of course!

**Comment by navendu-pottekkat** [comment]
The lint error is from a different PR. We can merge this if there are no issues.

**Comment by navendu-pottekkat** [comment]
I have fixed the link check error in a different PR. All checks will pass here.

**Review by tao12345666333 — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**